ClientFactory/CurlFactory.php

```diff
@@ -40,6 +40,18 @@ public function createClient(array $config = [])
             throw new \LogicException('To use the Curl client you need to install the "php-http/curl-client" package.');
         }
 
+        // Try to resolve curl constant names
+        foreach ($config as $key => $value) {
+            // If the value starts with 'CURL' we assume it is a reference to a constant.
+            if (strpos($key, 'CURL') === 0) {
+                $contantValue = constant($key);
```

Comment:
according to http://php.net/manual/en/function.curl-setopt.php it must be int. so either it already is an int here, or its a constant. maybe we could do

```
if (!is_int($key)) {
    $constantValue = constant($key);
    if (null === $constantValue) {
        throw... $key is not an integer or a CURL constant.
    }
    unset...,
    $config[$constantValue] = $value;
}
```